pdns/recursordist/settings/docs-new-preamble-in.rst

```diff
@@ -0,0 +1,185 @@
+      - '[::]:5301'
+  recursor:
+    extended_resolution_errors: true
+    forward_zones:
+      - zone: example.com
+        forwarders:
+          - 127.0.0.1:5301
+  outgoing:
+    query_local_address:
+      - 0.0.0.0
+      - '::'
+  logging:
+    loglevel: 6
+
+Take care when listing IPv6 addresses, as characters used for these are special to YAML.
+If in doubt, quote any string containing ``:``, ``[`` or ``]`` and use (online) tools to check your YAML syntax.
+Specify an empty sequence using ``[]``.
+
+The main setting file is called ``recursor.yml`` and will be processed first.
+This settings file might refer to other files via the `recursor.include_dir`_ setting.
+The next section will describe how settings specified in multiple files are merged.
+
+Merging multiple setting files
+------------------------------
+If `recursor.include_dir`_ is set, all ``.yml`` files in it will be processed in alphabetical order, modifying the  settings processed so far.
+
+For simple values like an boolean or number setting, a value in the processed file will overwrite an existing setting.
+
+For values of type sequence, the new value will *replace* the existing value if the existing value is equal to the ``default`` or if the new value is marked with the ``!override`` tag.
+Otherwise, the existing value will be *extended* with the new value by appending the new sequence to the existing.
+
+For example, with the above example ``recursor.yml`` and an include directory containing a file ``extra.yml``:
+
+.. code-block:: yaml
+
+  dnssec:
+    log_bogus: false
+  recursor:
+    forward_zones:
+      - zone: example.net
+        forwarders:
+          - '::1'
+  outgoing:
+     query_local_address: !override
+       - 0.0.0.0
+     dont_query: []
+
+After merging, ``dnssec.log_bogus`` will be ``false``, the sequence of ``recursor.forward_zones`` will contain 2 zones and the ``outgoing`` addresses used will contain one entry, as the ``extra.yml`` entry has overwritten the existing one.
+
+``outgoing.dont-query`` has a non-empty sequence as default value. The main ``recursor.yml`` did not set it, so before processing ``extra.yml`` had the default value.
+After processing ``extra.yml`` the value will be set to the empty sequence, as existing default values are overwritten by new values.
+
+.. warning::
+   The merging process does not process values deeper than the second level.
+   For example if the main ``recursor.yml`` specified a forward zone
+
+   .. code-block:: yaml
+
+     forward_zones:
+       - zone: example.net
+         forwarders:
+           - '::1'
+
+   and another settings file contains
+
+   .. code-block:: yaml
+
+     forward_zones:
+       - zone: example.net
+         forwarders:
+           - '::2'
+
+   The result will *not* be a a single forward with two IP addresses, but two entries for ``example.net``.
+   It depends on the specific setting how the sequence is processed further.
+   In the future we might add a check for this case.
```

Comment:
probably makes sense to open separate tickets for each promise or TODO in this PR